sdk/storage/Azure.Storage.Files.Shares/perf/Azure.Storage.Files.Shares.Perf/Scenarios/DownloadFile.cs

```diff
@@ -0,0 +1,120 @@
+﻿//Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Test.Perf;
+
+namespace Azure.Storage.Files.Shares.Perf.Scenarios
+{
+    /// <summary>
+    /// The performance test scenario focused on downloading files from the Azure Files Shares storage.
+    /// </summary>
+    /// <seealso cref="Azure.Test.Perf.PerfTest{SizeOptions}" />
+    public sealed class DownloadFile : PerfTest<SizeOptions>
+    {
+        /// <summary>
+        /// Access manager to the Azure storage Share.
+        /// This is shared across all instances of the test run.
+        /// </summary>
+        private static ShareClientManager s_shareClientManager;
+
+        /// <summary>
+        /// The path of the local file uploaded to Azure storage.
+        /// This is shared across all instances of the test run.
+        /// </summary>
+        private static string s_uploadFilePath;
+
+        /// <summary>
+        /// Name of the <see cref="ShareFileClient"/> where the file is uploaded.
+        /// This is shared across all instances of the test run.
+        /// </summary>
+        private static string s_fileClientName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DownloadFile"/> class.
+        /// </summary>
+        /// <param name="options">The set of options to consider for configuring the scenario.</param>
+        public DownloadFile(SizeOptions options) : base(options)
+        {
+        }
+
+        /// <summary>
+        /// Instantiates and creates a <see cref="ShareClient"/> and a <see cref="ShareDirectoryClient"/> for the test.
+        /// Also, creates a local file and uploads it to Azure File storage.
+        /// </summary>
+        public override async Task GlobalSetupAsync()
+        {
+            await base.GlobalSetupAsync();
+
+            s_shareClientManager = new ShareClientManager();
```

Comment:
We are initializing a static property in a non-static method. Does the property have to be static?

Reply:
`GlobalSetupAsync()` is run once across all parallel runs (and intialization of the test class). So, it is like a static constructor.
And yes, we want to share the same manager instance across all runs.

Reply:
Would an instance of DownloadFile be created for every parallel run? Which instance is the `GlobalSetupAsync` executed on? The first one?

Reply:
Yes, an instance is created for every parallel run. The method is called on the [first one created](https://github.com/Azure/azure-sdk-for-net/blob/f4d532ec684ac4a8d5b24c51afd7c44cd34f9441/common/Perf/Azure.Test.Perf/PerfProgram.cs#L101).

Reply:
Got it. Thank you for the explanation.